src/scanner.rs

```diff
@@ -217,7 +265,7 @@ impl Scanner {
             self.dedupe_by_content(fileset, path, metadata)?;
         } else {
             self.stats.hardlinks += 1;
-            self.stats.bytes_saved_by_hardlinks += metadata.size() as usize;
+            self.stats.bytes_saved_by_hardlinks += metadata.len() as usize;
```

Comment:
`size` was intentional, because removing file content also removes the block overhead.